### PR TITLE
async dialog content => async create dialog

### PR DESCRIPTION
**Pull request by le-jeu**
Wait for dialog content before showing the dialog
This allows jquery/iitc dialog to place the dialog in the center of the screen with respect to the content on opening.

Additionally, don't use Promise.all when all promises are already fulfilled (synchronously)

### CODE REVIEW

File: src/code/sortable.js [hunk 1]
```diff
@@ -105,23 +106,31 @@ export default class Sortable {
...
-        this._items = values;
-        this.sort();
-        return true;
-      },
-      (reject) => {
-        console.log("rejected", reject);
-        this._done = false;
-      }
-    );
+    if (instantValues.length < promises.length) {
```
Comment (cloudkucooland) [inline src/code/sortable.js:116]: It's a style thing, but I'd invert the logic of this 

if (IinstantValues.length == promises.length) {
 ...
 return
 }
 
 ...
 
 and my comment from back in the day says I wanted to try allSettled, did you look into that?

File: src/code/dialogs/markerList.js [hunk 2]
```diff
@@ -27,6 +27,8 @@ const MarkerList = OperationChecklistDialog.extend({
       this.closeDialog();
     };
 
+    await this.sortable.done;
```
Comment (cloudkucooland) [inline src/code/dialogs/markerList.js:30]: done is a getter, which isn't async. does this work?

**Review by cloudkucooland — COMMENTED** [review]
I'm not sure this works the way intended (or I read as intended)

File: src/code/sortable.js (hunk 1 above)
Comment (le-jeu) [reply src/code/sortable.js:116]: > It's a style thing, but I'd invert the logic of this

fine

> and my comment from back in the day says I wanted to try allSettled, did you look into that?

no, while it's not fail safe, the point of failure are to be defined first (include how to settle them)
atm, if one row fails, the issue is not only the rendering of the row but login/perm/team

File: src/code/dialogs/markerList.js (hunk 2 above)
Comment (le-jeu) [reply src/code/dialogs/markerList.js:30]: done returns _done that is either a promise or  boolean
also, you can always await a constant expression

**Review by cloudkucooland — APPROVED** [review]

**Pull request merged**